pkg/zones/wellknown.go

```diff
@@ -386,10 +391,52 @@ func GuessCloudForZone(zone string) (kops.CloudProviderID, bool) {
 	return matches[0], true
 }
 
-func WellKnownZonesForCloud(matchCloud kops.CloudProviderID) []string {
+func WellKnownZonesForCloud(matchCloud kops.CloudProviderID, prefix string) []string {
 	switch matchCloud {
 	case kops.CloudProviderAWS:
-		return awsZones
+		prefix = strings.ToLower(prefix)
+		awsCloud, err := awsup.NewAWSCloud("us-east-1", map[string]string{})
```

Comment:
We'll need to figure out a way to have this work for other AWS partitions

Reply:
I don't have the ability to test other partitions.

Reply:
I can test govcloud. We just need a way to use a partition-specific region. This would likely require parsing the resulting ARN from sts.GetCallerIdentity and using a map of partitions to region. It may be wise to respect AWS_REGION or any region in the AWS SDK's config lookup sequence as well.

[endpoints.DefaultPartitions](https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/) and their Regions() method might be able to help too.

Reply:
We could probably just query one region in each partition.

I don't think we want to limit to AWS_REGION. I see that as a default, not a restriction.

Reply:
Rewrote to use `endpoints.DefaultPartitions` to get the list of region names.